astroquery/jplhorizons/core.py

```diff
@@ -64,35 +69,43 @@ def __init__(self, id=None, location=None, epochs=None,
             the type of query performed: UTC for ephemerides queries, TDB for
             element and vector queries. If no epochs are provided, the current
             time is used.
+
         id_type : str, optional
             Controls Horizons's object selection for ``id``
             [HORIZONSDOC_SELECTION]_ .  Options: ``'designation'`` (small body
             designation), ``'name'`` (asteroid or comet name),
-            ``'asteroid_name'``, ``'comet_name'``, ``'smallbody'`` (asteroid
-            and comet search), or ``None`` (first search search planets,
-            natural satellites, spacecraft, and special cases, and if no
-            matches, then search small bodies).
+            ``'asteroid_name'``, ``'comet_name'``, ``'smallbody'`` (asteroid and
+            comet search), or ``None`` (first search search planets, natural
+            satellites, spacecraft, and special cases, and if no matches, then
+            search small bodies).
+
 
         References
         ----------
 
-        .. [HORIZONSDOC_SELECTION] https://ssd.jpl.nasa.gov/horizons/manual.html#select (retrieved 2021 Sep 23).
+        .. [HORIZONSDOC_SELECTION] https://ssd.jpl.nasa.gov/horizons/manual.html#select
+            (retrieved 2021 Sep 23).
 
 
         Examples
         --------
+
         >>> from astroquery.jplhorizons import Horizons
         >>> eros = Horizons(id='433', location='568',
-        ...              epochs={'start':'2017-01-01',
-        ...                      'stop':'2017-02-01',
-        ...                      'step':'1d'})
-        >>> print(eros)  # doctest: +SKIP
-        JPLHorizons instance "433"; location=568, epochs={'start': '2017-01-01', 'step': '1d', 'stop': '2017-02-01'}, id_type=None
+        ...                 epochs={'start': '2017-01-01',
+        ...                         'stop': '2017-02-01',
+        ...                         'step': '1d'})
+        >>> print(eros)
+        JPLHorizons instance "433"; location=568, epochs={'start': '2017-01-01', 'stop': '2017-02-01', 'step': '1d'}, id_type=None
+
         """
+
         super().__init__()
         self.id = id
         self.location = location
 
+        self.server_url = conf.horizons_server
```

Comment:
Currently  (in `main`) the configuration item is accessed in the functions that need the value, which means that changing it at runtime works as expected. With this change it would be accessed only when the `HorizonsClass` instance is created, so any changes to the configuration item after that would have no effect. That would be a regression.

Reply:
Then I am confused, because in #2358 @bsipocz said "hard-wiring it in the methods [(as opposed to initialization)] like this is very much unrecommended."

Reply:
I have updated the opening post of #2341 to explain the motivation behind the changes proposed there, bit I'll provide a short summary here.

The current behavior (in `main`) works just fine for everyone using the `astropy` configuration system, but not all users can be expected to be familiar with it. It would therefore be good if there were instance attributes that could be used as an _alternative_ to the configuration system, but the way da856a5d68f513e56960434c10fcbbbd1671f921 implements the instance attribute would mean that any changes made to the configuration system after instance creation would have no effect. That would be an obvious regression.

The best way to implement configurable values would be to follow the template given in #2341.

Reply:
I understand the motivation, I was just confused by the conflicting advice.  Regardless, I'll revert this change.  Thanks for the review.

Reply:
Yes, the configuration system needs those clarifications from #2341 (I'll get to the review of it), my point here (in fact not here, but in the new module) was to not hardwire the URL in each and every method of a class, have it set once on the class instance and call it a day for now. 
OTOH refactorings could wait for existing modules.